examples/classification/plot_classifier_comparison.py

```diff
@@ -106,9 +106,16 @@
     ax.set_yticks(())
     i += 1
 
-    # iterate over classifiers
-    for name, clf in zip(names, classifiers):
-        ax = plt.subplot(len(datasets), len(classifiers) + 1, i)
+# iterate over classifiers
+for clf_cnt, (name, clf) in enumerate(zip(names, classifiers)):
+    for ds_cnt, ds in enumerate(datasets):
+        # preprocess dataset, split into training and test part
+        X, y = ds
+        X = StandardScaler().fit_transform(X)
+        X_train, X_test, y_train, y_test = \
+            train_test_split(X, y, test_size=.4, random_state=42)
+
+        ax = plt.subplot(len(classifiers) + 1, len(datasets), i)
```

Comment:
can't you do the same thing here with constructing plot_num?

Another way to do this is use pyplot.subplots which returns an array of axes, which you can then fill in by accessing each cell's coordinates (or by iterating over the matrix, or its transpose).

Reply:
Sure, I'll use pyplot.subplots then, seems more explicit than plot_num